GopherJS incorrectly uses stale archive files when latest source code fails to build.

1. Create the following three files.
2. Run `gopherjs serve` and load `alltom/` in your browser.
3. Edit iface.go to change `Bar() int` to `Bar2() int`.
4. Edit main.go to change `impl.MakeFoo().Bar()` to `impl.MakeFoo().Bar2()`.
5. Reload your browser.

Instead of seeing a compile error about FooImpl no longer implementing Foo, you get a runtime error saying "Uncaught TypeError: impl.MakeFoo(...).Bar2 is not a function". It looks like `gopherjs serve` doesn't realize that impl.go needs to be recompiled.

## alltom/iface/iface.go
```go
package iface

type Foo interface {
	Bar() int
}
```

## alltom/impl/impl.go
```go
package impl

import "alltom/iface"

type FooImpl int

func MakeFoo() iface.Foo {
	return FooImpl(0)
}

func (f FooImpl) Bar() int {
	return 42
}
```

## alltom/main.go
```go
package main

import "alltom/impl"

func main() {
	impl.MakeFoo().Bar()
}
```

I can reproduce. It also happens with `gopherjs build` or `gopherjs run`, not just `gopherjs serve`.

Deleting `impl.a` from `GOPATH/pkg/darwin_js/alltom/impl.a` makes it produce correct compile error:

```
$ gopherjs run main.go 
impl/impl.go:8:9: cannot use FooImpl(0) (constant 0 of type FooImpl) as alltom/iface.Foo value in return statement: missing method Bar2
```

Since there's a compilation error, that `impl.a` doesn't get created. But if it already exists from a previous successful compilation run, it gets incorrectly used despite being stale.

It appears the problem is something like this. When a built file is stale, GopherJS correctly tries to rebuild it. However, if that package fails to build due to errors, it problematically still goes ahead and uses the stale file.

This is a bug in the build logic.

--- a/build/build.go
+++ b/build/build.go
@@ -532,11 +532,11 @@ func (s *Session) BuildPackage(pkg *PackageData) (*compiler.Archive, error) {
 			if importedPkgPath == "unsafe" || ignored {
 				continue
 			}
-			pkg, _, err := s.buildImportPathWithSrcDir(importedPkgPath, pkg.Dir)
+			importedPkg, _, err := s.buildImportPathWithSrcDir(importedPkgPath, pkg.Dir)
 			if err != nil {
 				return nil, err
 			}
-			impModeTime := pkg.SrcModTime
+			impModeTime := importedPkg.SrcModTime
 			if impModeTime.After(pkg.SrcModTime) {
 				pkg.SrcModTime = impModeTime
 			}

